### PR TITLE
No cancel token if the action is immediately consumed by UIScheduler.

### PR DESCRIPTION
**Pull request by andersio**
Change of an Implementation detail.

#### Checklist
- ~Updated CHANGELOG.md.~

### CODE REVIEW

File: Sources/Scheduler.swift [hunk 1]
```diff
@@ -162,11 +159,17 @@ public final class UIScheduler: Scheduler {
...
 			actionAndDecrement()
+			return nil
 		} else {
-			DispatchQueue.main.async(execute: actionAndDecrement)
-		}
+			let disposable = AnyDisposable()
 
-		return disposable
+			DispatchQueue.main.async {
+				guard !disposable.isDisposed else { return }
```
Comment (sharplet) [inline Sources/Scheduler.swift]: Is this a change in behaviour? Should there be a new test for this?

Comment (andersio) [inline Sources/Scheduler.swift]: The only change is that `schedule` that is done immediately no longer returns a disposable, aligning to the semantic of `ImmediateScheduler`. Per the `Scheduler` contract, it is wrong to assume `schedule` would always return a cancellation token anyway.

Comment (andersio) [inline Sources/Scheduler.swift]: Argh, you catch a flaw I've made.

Comment (sharplet) [inline Sources/Scheduler.swift]: Specifically though, why was an explicit check of the disposable added?

Comment (andersio) [inline Sources/Scheduler.swift]: It was there before.
https://github.com/ReactiveCocoa/ReactiveSwift/pull/439/commits/04b4ddf4500a2f93dcbef286ace2d899c0918621#diff-f045af08e7f58fd005590d376ccd1215L141

**Pull request merged**